<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CWorldEditor</class>
 <widget class="QMainWindow" name="CWorldEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>720</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Prime World Editor</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_5">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="">
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="CSceneViewport" name="MainViewport" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>1</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QFrame" name="BottomFrame">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Plain</enum>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <property name="leftMargin">
            <number>3</number>
           </property>
           <property name="topMargin">
            <number>3</number>
           </property>
           <property name="rightMargin">
            <number>3</number>
           </property>
           <property name="bottomMargin">
            <number>3</number>
           </property>
           <item>
            <widget class="QFrame" name="SelectionInfoFrame">
             <property name="minimumSize">
              <size>
               <width>200</width>
               <height>0</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>200</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="frameShape">
              <enum>QFrame::Panel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Sunken</enum>
             </property>
             <property name="lineWidth">
              <number>1</number>
             </property>
             <property name="midLineWidth">
              <number>0</number>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_3">
              <property name="leftMargin">
               <number>3</number>
              </property>
              <property name="topMargin">
               <number>3</number>
              </property>
              <property name="rightMargin">
               <number>3</number>
              </property>
              <property name="bottomMargin">
               <number>3</number>
              </property>
              <item>
               <widget class="QLabel" name="SelectionInfoLabel">
                <property name="frameShadow">
                 <enum>QFrame::Plain</enum>
                </property>
                <property name="lineWidth">
                 <number>1</number>
                </property>
                <property name="midLineWidth">
                 <number>0</number>
                </property>
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="WVectorEditor" name="TransformSpinBox" native="true">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>300</width>
               <height>0</height>
              </size>
             </property>
            </widget>
           </item>
           <item>
            <widget class="Line" name="BottomFrameDivider">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QLabel" name="label">
               <property name="text">
                <string>Speed:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="WDraggableSpinBox" name="CamSpeedSpinBox">
               <property name="maximumSize">
                <size>
                 <width>50</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="focusPolicy">
                <enum>Qt::NoFocus</enum>
               </property>
               <property name="contextMenuPolicy">
                <enum>Qt::NoContextMenu</enum>
               </property>
               <property name="frame">
                <bool>true</bool>
               </property>
               <property name="correctionMode">
                <enum>QAbstractSpinBox::CorrectToNearestValue</enum>
               </property>
               <property name="keyboardTracking">
                <bool>false</bool>
               </property>
               <property name="minimum">
                <double>0.010000000000000</double>
               </property>
               <property name="maximum">
                <double>10.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.100000000000000</double>
               </property>
               <property name="value">
                <double>1.000000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <spacer name="BottomFrameSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QTabWidget" name="TabWidget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="baseSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="focusPolicy">
        <enum>Qt::NoFocus</enum>
       </property>
       <property name="tabPosition">
        <enum>QTabWidget::North</enum>
       </property>
       <property name="tabShape">
        <enum>QTabWidget::Rounded</enum>
       </property>
       <property name="currentIndex">
        <number>1</number>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
       <property name="documentMode">
        <bool>false</bool>
       </property>
       <widget class="QWidget" name="CreateTab">
        <attribute name="icon">
         <iconset resource="../Icons.qrc">
          <normaloff>:/icons/Create.png</normaloff>:/icons/Create.png</iconset>
        </attribute>
        <attribute name="title">
         <string/>
        </attribute>
        <attribute name="toolTip">
         <string>Create</string>
        </attribute>
       </widget>
       <widget class="QWidget" name="ModifyTab">
        <attribute name="icon">
         <iconset resource="../Icons.qrc">
          <normaloff>:/icons/Modify.png</normaloff>:/icons/Modify.png</iconset>
        </attribute>
        <attribute name="title">
         <string/>
        </attribute>
        <attribute name="toolTip">
         <string>Modify</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="WModifyTab" name="ModifyTabContents" native="true"/>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="InstancesTab">
        <attribute name="icon">
         <iconset resource="../Icons.qrc">
          <normaloff>:/icons/Instances.png</normaloff>:/icons/Instances.png</iconset>
        </attribute>
        <attribute name="title">
         <string/>
        </attribute>
        <attribute name="toolTip">
         <string>Instances</string>
        </attribute>
        <layout class="QHBoxLayout" name="InstancesTabLayout">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="WInstancesTab" name="InstancesTabContents" native="true"/>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="DisplayTab">
        <attribute name="icon">
         <iconset resource="../Icons.qrc">
          <normaloff>:/icons/Display.png</normaloff>:/icons/Display.png</iconset>
        </attribute>
        <attribute name="title">
         <string/>
        </attribute>
        <attribute name="toolTip">
         <string>Display</string>
        </attribute>
       </widget>
       <widget class="QWidget" name="WorldTab">
        <attribute name="icon">
         <iconset resource="../Icons.qrc">
          <normaloff>:/icons/World.png</normaloff>:/icons/World.png</iconset>
        </attribute>
        <attribute name="title">
         <string/>
        </attribute>
        <attribute name="toolTip">
         <string>World</string>
        </attribute>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1280</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="ActionOpen"/>
    <addaction name="ActionSave"/>
   </widget>
   <widget class="QMenu" name="menuWindow">
    <property name="title">
     <string>Window</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuModels">
    <property name="title">
     <string>Models</string>
    </property>
    <addaction name="actionOpen_model_viewer"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="ActionSelectAll"/>
    <addaction name="ActionInvertSelection"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <widget class="QMenu" name="menuLighting">
     <property name="title">
      <string>Lighting</string>
     </property>
     <addaction name="ActionNoLighting"/>
     <addaction name="ActionBasicLighting"/>
     <addaction name="ActionWorldLighting"/>
    </widget>
    <widget class="QMenu" name="menuBloom">
     <property name="title">
      <string>Bloom</string>
     </property>
     <addaction name="ActionNoBloom"/>
     <addaction name="ActionBloomMaps"/>
     <addaction name="ActionFakeBloom"/>
     <addaction name="ActionBloom"/>
    </widget>
    <addaction name="ActionGameMode"/>
    <addaction name="separator"/>
    <addaction name="ActionDrawWorld"/>
    <addaction name="ActionDrawObjects"/>
    <addaction name="ActionDrawCollision"/>
    <addaction name="ActionDrawObjectCollision"/>
    <addaction name="ActionDrawLights"/>
    <addaction name="ActionDrawSky"/>
    <addaction name="separator"/>
    <addaction name="menuLighting"/>
    <addaction name="menuBloom"/>
    <addaction name="separator"/>
    <addaction name="ActionDisableBackfaceCull"/>
    <addaction name="ActionDisableAlpha"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="ActionEditLayers"/>
    <addaction name="ActionEditPoiToWorldMap"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuTools"/>
   <addaction name="menuView"/>
   <addaction name="menuModels"/>
   <addaction name="menuWindow"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="FileToolBar">
   <property name="contextMenuPolicy">
    <enum>Qt::NoContextMenu</enum>
   </property>
   <property name="windowTitle">
    <string>toolBar_2</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="ActionOpen"/>
   <addaction name="ActionSave"/>
  </widget>
  <widget class="QToolBar" name="MainToolBar">
   <property name="contextMenuPolicy">
    <enum>Qt::NoContextMenu</enum>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="iconSize">
    <size>
     <width>32</width>
     <height>32</height>
    </size>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>true</bool>
   </attribute>
   <addaction name="ActionLink"/>
   <addaction name="ActionUnlink"/>
   <addaction name="separator"/>
  </widget>
  <action name="ActionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="ActionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionOpen_model_viewer">
   <property name="text">
    <string>Open Model Viewer</string>
   </property>
  </action>
  <action name="ActionLink">
   <property name="icon">
    <iconset resource="../Icons.qrc">
     <normaloff>:/icons/Link.png</normaloff>:/icons/Link.png</iconset>
   </property>
   <property name="text">
    <string>Link</string>
   </property>
   <property name="toolTip">
    <string>Link</string>
   </property>
  </action>
  <action name="ActionUnlink">
   <property name="icon">
    <iconset resource="../Icons.qrc">
     <normaloff>:/icons/Unlink.png</normaloff>:/icons/Unlink.png</iconset>
   </property>
   <property name="text">
    <string>Unlink</string>
   </property>
   <property name="toolTip">
    <string>Unlink</string>
   </property>
  </action>
  <action name="ActionDrawWorld">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>World</string>
   </property>
   <property name="shortcut">
    <string>1</string>
   </property>
  </action>
  <action name="ActionDrawCollision">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Collision</string>
   </property>
   <property name="shortcut">
    <string>3</string>
   </property>
  </action>
  <action name="ActionDrawObjects">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Objects</string>
   </property>
   <property name="shortcut">
    <string>2</string>
   </property>
  </action>
  <action name="ActionDrawLights">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Lights</string>
   </property>
   <property name="shortcut">
    <string>5</string>
   </property>
  </action>
  <action name="ActionDrawSky">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Sky</string>
   </property>
   <property name="shortcut">
    <string>6</string>
   </property>
  </action>
  <action name="ActionNoLighting">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>None</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+1</string>
   </property>
  </action>
  <action name="ActionBasicLighting">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Basic</string>
   </property>
   <property name="toolTip">
    <string>Basic</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+2</string>
   </property>
  </action>
  <action name="ActionWorldLighting">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>World</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+3</string>
   </property>
  </action>
  <action name="ActionNoBloom">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>None</string>
   </property>
  </action>
  <action name="ActionBloomMaps">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Bloom Maps</string>
   </property>
  </action>
  <action name="ActionBloom">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Bloom</string>
   </property>
  </action>
  <action name="ActionZoomOnSelection">
   <property name="text">
    <string>Zoom On Selection</string>
   </property>
   <property name="shortcut">
    <string>Z</string>
   </property>
  </action>
  <action name="ActionDisableBackfaceCull">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Disable Backface Culling</string>
   </property>
  </action>
  <action name="ActionDisableAlpha">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Disable Alpha</string>
   </property>
  </action>
  <action name="ActionEditLayers">
   <property name="text">
    <string>Edit Layers</string>
   </property>
  </action>
  <action name="ActionFakeBloom">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Fake Bloom</string>
   </property>
  </action>
  <action name="ActionIncrementGizmo">
   <property name="text">
    <string>Increment Gizmo Size</string>
   </property>
   <property name="toolTip">
    <string>Increment Gizmo Size</string>
   </property>
   <property name="shortcut">
    <string>=</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::WindowShortcut</enum>
   </property>
  </action>
  <action name="ActionDecrementGizmo">
   <property name="text">
    <string>Decrement Gizmo Size</string>
   </property>
   <property name="toolTip">
    <string>Decrement Gizmo Size</string>
   </property>
   <property name="shortcut">
    <string>-</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::WindowShortcut</enum>
   </property>
  </action>
  <action name="ActionDrawObjectCollision">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Object Collision</string>
   </property>
   <property name="shortcut">
    <string>4</string>
   </property>
  </action>
  <action name="ActionGameMode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Game Mode</string>
   </property>
   <property name="shortcut">
    <string>G</string>
   </property>
  </action>
  <action name="ActionSelectAll">
   <property name="text">
    <string>Select All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="ActionInvertSelection">
   <property name="text">
    <string>Invert Selection</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="ActionEditPoiToWorldMap">
   <property name="text">
    <string>Edit POI to World Map</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>WDraggableSpinBox</class>
   <extends>QDoubleSpinBox</extends>
   <header>Editor/Widgets/WDraggableSpinBox.h</header>
  </customwidget>
  <customwidget>
   <class>WModifyTab</class>
   <extends>QWidget</extends>
   <header>Editor/WorldEditor/WModifyTab.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>WInstancesTab</class>
   <extends>QWidget</extends>
   <header>Editor/WorldEditor/WInstancesTab.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>WVectorEditor</class>
   <extends>QWidget</extends>
   <header>Editor/Widgets/WVectorEditor.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>CSceneViewport</class>
   <extends>QWidget</extends>
   <header>Editor/CSceneViewport.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../Icons.qrc"/>
 </resources>
 <connections/>
</ui>
